mm2src/coins/utxo/slp.rs

```diff
@@ -129,6 +130,14 @@ impl From<UnexpectedDerivationMethod> for ValidateHtlcError {
     fn from(e: UnexpectedDerivationMethod) -> Self { ValidateHtlcError::UnexpectedDerivationMethod(e) }
 }
 
+impl From<ValidateHtlcError> for ValidatePaymentError {
+    fn from(e: ValidateHtlcError) -> Self { ValidatePaymentError::ValidateHtlcError(e.to_string()) }
+}
+
+impl From<ValidatePaymentError> for ValidateHtlcError {
+    fn from(e: ValidatePaymentError) -> Self { ValidateHtlcError::ValidatePaymentError(e.to_string()) }
+}
```

Comment:
Such cyclic conversion looks weird.
An actual situation is quite difficult to resolve.
`slp::validate_maker_payment` uses `slp::validate_htlc`, but it in turn uses `utxo_common::validate_payment`.

Please try to remove `ValidateHtlcError` by returning `ValidatePaymentError` from `slp::validate_htlc`.
For example, `validate_slp_utxos() -> ValidateSlpUtxosErr` error should be converted into `ValidatePaymentError::InvalidPaymentTxData` with the corresponding error message.

Reply:
Alright noted and done 👍